### PR TITLE
Fix few missing memory region names on stm32 boards

### PR DESCRIPTION
**Pull request by fabiobaltieri**
Hey hi, found few missing region names after https://github.com/zephyrproject-rtos/zephyr/pull/37279 on SDRAM and CCM nodes for some stm32 boards. One had no compatible, the other missed the binding, tested with `west build -p -b stm32f429i_disc1 samples/basic/blinky`.

Before:

```
Memory region         Used Size  Region Size  %age Used
           FLASH:       14988 B         2 MB      0.71%
            SRAM:        4320 B       192 KB      2.20%
/memory@10000000:          0 GB        64 KB      0.00%
 /sdram@d0000000:          0 GB         8 MB      0.00%
        IDT_LIST:          0 GB         2 KB      0.00%
```

after:

```
Memory region         Used Size  Region Size  %age Used
           FLASH:       14988 B         2 MB      0.71%
            SRAM:        4320 B       192 KB      2.20%
             CCM:          0 GB        64 KB      0.00%
          SDRAM2:          0 GB         8 MB      0.00%
        IDT_LIST:          0 GB         2 KB      0.00%
```

Also fix the property name used in `zephyr_linker_dts_memory`, right now it's always using the fallback.

@JordanYates

### CODE REVIEW

**Review by FRASTM — APPROVED** [review]

File: dts/bindings/memory-controllers/st,stm32-fmc-sdram.yaml [hunk 1]
```diff
@@ -56,11 +56,13 @@ description: |
   memory device/s in DeviceTree:
 
   sdram1: sdram@c0000000 {
+      compatible = "mmio-sram";
       device_type = "memory";
       reg = <0xc000000 DT_SIZE_M(X)>;
   };
```
Comment (erwango) [inline dts/bindings/memory-controllers/st,stm32-fmc-sdram.yaml:62]:
```suggestion
+         zephyr,memory-region = "SDRAM1";
+};
```

File: dts/bindings/memory-controllers/st,stm32-fmc-sdram.yaml [hunk 2]
```diff
@@ -56,11 +56,13 @@ description: |
...
 
   sdram2: sdram@d0000000 {
+      compatible = "mmio-sram";
       device_type = "memory";
       reg = <0xd000000 DT_SIZE_M(X)>;
   };
```
Comment (erwango) [inline dts/bindings/memory-controllers/st,stm32-fmc-sdram.yaml:68]:
```suggestion
+         zephyr,memory-region = "SDRAM2";
+};
```

**Review by erwango — CHANGES_REQUESTED** [review]
Thanks @fabiobaltieri. Only minor comments

File: boards/arm/stm32f769i_disco/stm32f769i_disco.dts [hunk 3]
```diff
@@ -23,8 +23,10 @@
 	};
 
 	sdram1: sdram@c0000000 {
+		compatible = "mmio-sram";
 		device_type = "memory";
```
Comment (erwango) [inline boards/arm/stm32f769i_disco/stm32f769i_disco.dts:27]: Looking at this (and not related to this PR):
@mbolivar-nordic: device_type prop is nowhere documented in bindings. Shouldn't it be removed ?

File: dts/bindings/memory-controllers/st,stm32-fmc-sdram.yaml (hunk 1 above)
Comment (fabiobaltieri) [reply dts/bindings/memory-controllers/st,stm32-fmc-sdram.yaml:63]: Right!!

File: dts/bindings/memory-controllers/st,stm32-fmc-sdram.yaml (hunk 2 above)
Comment (fabiobaltieri) [reply dts/bindings/memory-controllers/st,stm32-fmc-sdram.yaml:70]: Fixed.

File: boards/arm/stm32f769i_disco/stm32f769i_disco.dts (hunk 3 above)
Comment (fabiobaltieri) [reply boards/arm/stm32f769i_disco/stm32f769i_disco.dts:27]: Curious as well, shows up in `edtlib.py` but I don't think it's used anywhere and it's declared as deprecated in the specs https://github.com/devicetree-org/devicetree-specification/blob/main/source/chapter2-devicetree-basics.rst#device_type-deprecated.

**Review by tejlmand — APPROVED** [review]

**Review by erwango — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

File: boards/arm/stm32f769i_disco/stm32f769i_disco.dts (hunk 3 above)
Comment (mbolivar-nordic) [reply boards/arm/stm32f769i_disco/stm32f769i_disco.dts:27]: > it's declared as deprecated in the specs https://github.com/devicetree-org/devicetree-specification/blob/main/source/chapter2-devicetree-basics.rst#device_type-deprecated.

@fabiobaltieri the link you pasted says (emphasis mine):

>  new use of the property is deprecated, and **it should be included only on cpu and memory nodes** for compatibility with IEEE 1275–derived devicetrees.

So from a specification point of view, it seems fine to keep this as is.

@galak asked about this on the spec mailing list, but the results were inconclusive: https://www.spinics.net/lists/devicetree-spec/msg00800.html

Take from that what you will.

**Pull request merged**